webpack.config.js

```diff
@@ -16,19 +16,33 @@ module.exports = {
   plugins: [
     new webpack.DefinePlugin({
       'process.env': {
-        'NODE_ENV': JSON.stringify('production')
+        NODE_ENV: JSON.stringify('production')
       }
     })
   ],
   module: {
     rules: [
-      { test: /\.js?$/, use: ['babel-loader', 'eslint-loader'], exclude: /node_modules/ },
+
+      // All files with a '.ts' or '.tsx' extension will be handled by 'ts-loader'.
+      {
+        test: /\.tsx?$/,
+        use: 'ts-loader',
+        exclude: /node_modules/
+      },
+
+      // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
+      { test: /\.js$/, loader: 'source-map-loader' },
```

Comment:
Is this so the source maps can point to the *.ts files? If so, do source maps work for you so far? I think we need to include the source code in the NPM package before the source maps can work (see .npmignore)

Reply:
Honestly, didn't use or test this so far. Was just a copy&paste from a Typescript/Webpack guide.
I'll keep this open here until we have a better understanding, what needs to be included, such that debugging is possible for users of the package.

Reply:
For the first time, I tried to run ecamp3 with this branch of `hal-json-vuex`.

**The bad thing:** Something is clearly not working properly, error in console & no data loaded at all

**The good thing:** The error message in the browser directly lead me to to the correct source code line in `index.ts`. So the source map seems to work. Looks like `bundle.js.map` includes the complete original Typescript code.

Reply:
Nice. How did you run it with this branch? Did you `npm run build` and then copy only the files from `dist` over to ecamp3?

Reply:
Sort of, but more automated. I tried first with `npm link`, but docker obviosuly doesn't like it because the symlink will be outside of docker's volume.

To the rescue:
https://github.com/whitecolor/yalc

In `hal-json-vuex`:
```
npm run build
sudo yalc publish
```

in `ecamp3/frontend`:
```
yalc add hal-json-vuex
```

and later
```
yalc update
```

I'm quite sure I managed to run it with `yalc link hal-json-vuex` earlier. But somehow it doesn't work anymore. Theoritically it should work because the symlink stays within `frontend` and therefore within the docker volume.